doc/scss/custom.scss

```diff
@@ -190,3 +190,54 @@ div.sk-text-image-grid-small {
 div.sk-text-image-grid-large {
   @include sk-text-image-grid(100px);
 }
+
+/* Style of the docsearch modal window */
+
+// In the following, we mainly reuse the background color and font-color from the
+// pydata-sphinx-theme. We also adjust slightly the font-size to be wider and in line
+// with the sidebar design.
+:root {
+  --docsearch-modal-width: 768px;
+  --docsearch-modal-height: 700px;
+  --docsearch-searchbox-shadow: inset 0 0 0 2px var(--pst-color-primary);
+  --docsearch-highlight-color: var(--pst-color-primary);
+}
+
```

Comment:
Maybe this?

```scss
@media screen and (max-width: 1200px) {
  .DocSearch-Button-Placeholder,
  .DocSearch-Button-Keys {
    display: none;
  }
}
```

Currently the placeholder text and the keyboard shortcut hints are hidden only on mobile size. However, at width `1023px` the top navbar already cannot hold all these items.

![image](https://github.com/scikit-learn/scikit-learn/assets/108576690/fe0d017a-f199-41a6-bc4f-9f0800743e04)

The `1200px` breakpoint is at which the secondary sidebar is folded.